ALEC-228: Create Quick Start section (#97)
OpenNMS/alec: docs/modules/quick-start/pages/engine-config.adoc

:imagesdir: ../assets/images
= Step 1: First-Time Login and Engine Configuration

This section explains first-time login tasks and how to configure a correlation engine.

> ALEC includes two correlation engines: Clustering and Deep Learning.

Reviewer: ```suggestion
ALEC includes two correlation engines: clustering and deep learning.
```

> The engine that you choose determines how alerts are grouped into clusters.

Reviewer: ```suggestion
The engine that you choose determines how ALEC groups alerts.
```
Is it important they are grouped into clusters, or just that they are grouped? Does "cluster" mean the same as "situation" or is there a distinction we need to make here?

Author: Clusters, from what I understand, are the prerequisite for situations. Clusters simply group alarms together; situations represent a synthetic "parent" alarm (as listed in the glossary). I'll get some clarification before we go forward.

> This Quick Start guide explains how to set up and configure the xref:engines:clustering.adoc[Clustering] engine.

Reviewer: ```suggestion
This Quick Start guide explains how to set up and configure the xref:engines:clustering.adoc[clustering] engine.
```

> For information on the Deep Learning engine, see xref:engines:deeplearning.adoc[Deep Learning].

Reviewer: ```suggestion
The xref:engines:deeplearning.adoc[deep learning] engine requires additional setup, and is beyond the scope of this Quick Start.
```
Something like this to indicate why we aren't covering it here.
We should confirm if anything happens if they do select deep learning and they haven't installed the dl engine. Do they get an error message? Should this even be an option if they don't have it installed?

After you install ALEC, you can access it via the OpenNMS web UI.
Log in to the web UI and click *Plugins > ALEC* in the top menu bar.
An introduction to the ALEC plugin is displayed, explaining the plugin and its key benefits.

== First-time login and data choices

As part of the introduction to ALEC, OpenNMS prompts for permission to send ALEC usage data to The OpenNMS Group.
We use this anonymized information to help determine product use and improve the ALEC plugin.
We do not share it with third parties, and will not use it for sales purposes.

.ALEC data permissions prompt
image::data-permissions.png[ALEC data permissions prompt, 750]

Select *Yes* or *No*, and click *Continue*.
The correlation engine configuration screen is displayed.

> == Configure Clustering engine

Reviewer: ```suggestion
== Configure clustering engine
```

When you access the ALEC plugin for the first time in the web UI, you are prompted to select a correlation engine.

.Correlation engine selection prompt
image::engine-config.png[Correlation engine selection prompt; includes details about the Clustering and Deep Learning correlation engines, 900]

Choose *Clustering*, and click *Continue*.

> The Clustering engine is now set.

Reviewer: ```suggestion
The clustering engine is now set.
```

> NOTE: After you select a correlation engine, it can be changed and updated only via the Karaf shell.

Reviewer: ```suggestion
NOTE: After you select a correlation engine, you can change and update it only via the Karaf shell.
```

:imagesdir: ../assets/images
= Step 1: First-Time Login and Engine Configuration

This section explains first-time login tasks and how to configure a correlation engine.
ALEC includes two correlation engines: clustering and deep learning.
The engine that you choose determines how ALEC groups alerts into clusters.

This Quick Start guide explains how to set up and configure the xref:engines:clustering.adoc[clustering] engine.

> The xref:engines:deeplearning.adoc[deep learning] engine requires additional setup, and is beyond the scope of this Quick Start guide.

Reviewer: ```suggestion
The xref:engines:deeplearning.adoc[deep learning] engine requires additional setup, and is covered as an advanced topic in the administration section.
```
Walking back my previous edit, as we have Quick Start twice in the same paragraph. 
We can add a link to the topic when we merge the installation PR.

After you install ALEC, you can access it via the OpenNMS web UI.
Log in to the web UI and click *Plugins > ALEC* in the top menu bar.
An introduction to the ALEC plugin is displayed, explaining the plugin and its key benefits.

== First-time login and data choices

> As part of the introduction to ALEC, OpenNMS prompts for permission to send ALEC usage data to The OpenNMS Group.

Reviewer: ```suggestion
When you first log in and navigate to the ALEC plugin, OpenNMS prompts for permission to send usage data to The OpenNMS Group.
```
